### PR TITLE
feat(server): implementing hono adapter

### LINKED ISSUE #1270: [Feature Request] Honojs server adapter support

### PR DESCRIPTION
**Pull request by svetch**
Fixes #1270

### CODE REVIEW

File: packages/server/src/hono/handler.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    return async (ctx) => {
+        const prisma = (await options.getPrisma(ctx)) as DbClientContract;
+        if (!prisma) {
+            return ctx.json({ message: 'unable to get prisma from request context' }, 500);
+        }
+
+        const url = new URL(ctx.req.url);
+        const query = Object.fromEntries(url.searchParams);
+
+        const path = ctx.req.path.substring(ctx.req.routePath.length - 1);
```
[the remark this answers is not in the document]
Comment (ymc9) [reply packages/server/src/hono/handler.ts:31]: Thanks for submitting the PR @svetch ! I've left a comment there, and the rest looks great. I especially appreciate the test cases 😄.

The usual practice we follow is to only include features in minor or major releases. So I'll hold this PR for a while for v2.7.0.

[the remark this answers is not in the document]
Comment (svetch) [reply packages/server/src/hono/handler.ts:31]: > Is this for trimming the trailing slash? I'm wondering if Hono guarantees there will be a trailing slash. Maybe we should explicitly check it?

This part is for trimming the route path, like '/api/*', which is passed into the middleware where it is used.

Should we trim the trailing slash? I didn't even think about that.

**Comment by LilaRest** [comment]
Hey guys! When should we expect the 2.7.0 to be released?

File: packages/server/src/hono/handler.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ymc9) [reply packages/server/src/hono/handler.ts:31]: Oh, I see, I misunderstood it. No, there shouldn't be a need to trim the trailing slash. Looking good to me now!

**Comment by ymc9** [comment]
> Hey guys! When should we expect the 2.7.0 to be released?



> Hey guys! When should we expect the 2.7.0 to be released?

Hi @LilaRest , I hope around 10/15. The adapter code is actually self contained, so if you are waiting for it, you can copy paste into your code base. When it's released, simply remove the code and import from `@zenstackhq/server/hono`.

Also, it'll be great if you could help test the code. I'm not experienced with Hono at all 😄

**Comment by LilaRest** [comment]
@ymc9 Alright sounds great, thanks! I'll do that way 😄

**Comment by svetch** [comment]
> @svetch , I think we need to add an export entry for hono in "package.json" file as well.

This is what I forgot but I fixed it quickly .:)

**Pull request merged**